src/extensions/surveys.tsx

```diff
@@ -187,18 +195,38 @@ export class SurveyManager {
 
         const surveySeen = getSurveySeen(survey)
         if (!surveySeen) {
+            this.clearSurveyTimeout(survey.id)
             this.addSurveyToFocus(survey.id)
+            const delaySeconds = survey.appearance?.surveyPopupDelaySeconds || 0
             const shadow = createShadow(style(survey?.appearance), survey.id, undefined, this.posthog)
-            Preact.render(
-                <SurveyPopup
-                    key={'popover-survey'}
-                    posthog={this.posthog}
-                    survey={survey}
-                    removeSurveyFromFocus={this.removeSurveyFromFocus}
-                    isPopup={true}
-                />,
-                shadow
-            )
+            if (delaySeconds <= 0) {
+                return Preact.render(
+                    <SurveyPopup
+                        key={'popover-survey'}
+                        posthog={this.posthog}
+                        survey={survey}
+                        removeSurveyFromFocus={this.removeSurveyFromFocus}
+                        isPopup={true}
+                    />,
+                    shadow
+                )
+            }
+            const timeoutId = setTimeout(() => {
+                if (!doesSurveyUrlMatch(survey)) {
+                    return this.removeSurveyFromFocus(survey.id)
+                }
+                Preact.render(
+                    <SurveyPopup
+                        key={'popover-survey'}
+                        posthog={this.posthog}
+                        survey={{ ...survey, appearance: { ...survey.appearance, surveyPopupDelaySeconds: 0 } }}
```

Comment:
why do we force `surveyPopupDelaySeconds: 0` here?

Reply:
because we're already handling the delay within the SurveyManager, so no need to handle it again in the SurveyPopup logic.

I'm still keeping the SurveyPopup logic for backwards compatibility, though shouldn't be used anymore by surveys managed by us (i.e. not triggered via `renderSurvey`)